### ISSUE TITLE
"My Activity Stream" dashlet: empty lines for inaccessible records

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#1491]
I'm running SuiteCRM 7.6.2. I am getting empty rows in the dashlet, like this:

![empty_rows](https://cloud.githubusercontent.com/assets/15945027/15551017/eb2fa14c-22ab-11e6-87bf-29acb372139b.jpg)

The lines that are missing at the bottom (blank), are lines that I can see when I log in with another user with more permissions.

**Those are lines relative to the creation of Contacts to which this user does not have access, in terms of Security Groups.**

In general, inaccessible records in SuiteCRM get **omitted** in lists, not shown as empty lines... 
This makes more sense: 
- in terms of **security** (not letting people know that "stuff" is happening, nly they don't have access to it), 
- and in terms of **usability** (some of my users are getting dozens of blank lines, instead of seeing the content that interests them).

Another user in the forums confirmed he was also seeing this behavior.

Maybe Jason @eggsurplus can help us out here. Thanks in advance!

**Comment by eggsurplus** [comment]
I had to do it this way due to how the SugarBean works. It would have meant ripping out a bunch of core code along with additional performance overhead. As this is now completely forked from CE, I don't see a reason why this can't be addressed going forward. Just wanted to add a little history behind what you are seeing.

**Comment by shogunpol** [comment]
@pgorod , issue has been tested on recent version SuiteCRM(7.7.8), and not appear. Can you confirm it and close an issue?

**Comment by pgorod** [comment]
@shogunpol I tried it now and I'm not seeing the bug. It seems to work fine now...

But this didn't fix itself spontaneously, did it???

**Comment by boombata** [comment]
Hi, 

Got this behavior on suitecrm 7.7.9.

the reason is that i'm working with sercurity groups. roles are set to GROUP so each security group users cannot see posts from the other group. however a blank line is added for each post from the other group. 

![scrm](https://cloud.githubusercontent.com/assets/8605061/21985845/fca62242-dbfc-11e6-8446-ba833f518a10.png)

**Comment by pgorod** [comment]
@boombata I would stay with this issue (reopen it), not open a new one. The information is here, the right participants are here, the new issue doesn't really help if it is exactly the same thing.

I don't think this is solved, unless somebody can point me to the commits that solved it. But I'm having trouble reproducing it now on my system.

I would say the way forward would be to really build up a list of steps to reproduce it, even if takes quite a few steps configuring groups, etc. Can you get us that, boombata?

@shogunpol it's your call about which issues to reopen/close.

**Comment by boombata** [comment]
@pgorodHere are my settings: 

1- role A
modules set to enabled and All 
uers: user1
security groups: none
2- role B
modules set to enabled and Group
uers: user2
security groups: none
3- security group A 
not inheritable checked
users: user1
role: none
4- security group B
not inheritable UNchecked
users: user2
role: none

5- To reproduce: 
- login with user1 
- Create a case with user1 
- login with user2 
- there should be a blank line in activity stream

**Comment by eggsurplus** [comment]
I'll take a look at this once more. I might have done something to address this issue since my May comment in the full edition of SecuritySuite. If so, I will merge in the change here.

**Comment by pgorod** [comment]
Thanks! I'm reopening this issue then.

**Comment by danieldias51** [comment]
Hi,

I had the same problem and I want to share the way I solve it:

On the file include/ListView/ListViewData.php on line 437 I add the following:
```
if($data[$dataIndex]['NAME'] == ""){
   unset($data[$dataIndex]);
}
```

I know it may not be the best way to solve the problem, but I just wanted to share my solution.
Btw, I'm using SuiteCRM version 7.7.8

**Comment by shogunpol** [comment]
@danieldias51 , i suggest create Pull Request on "Hotfix" branch so we can test it.

**Comment by pgorod** [comment]
This can be a nice workaround (thanks Daniel!), but is it our definitive solution to merge into the codebase? In terms of performance (and there can be hundreds of notifications on these lists), it's adding something to a list only to remove it afterwards.

I expect a change in the SecuritySuite code, to keep the entry from getting onto the array in the first place, would be much better. That code is already doing _something_ after checking security settings, because the entry gets cleared to an empty state. _That's_ the point where it should be kept from being appended to the array of notifications...

**Comment by danieldias51** [comment]
@pgorod I completely agree with you, this shouldn't be a definitive solution. This should be fixed properly.

I just shared the way I fixed it because could help others as a temporary workaround.

**Comment by cameronblaikie** [comment]
Closed due to inactivity, if you feel this should be reopened then do so and we will review your issue again.

**Comment by pgorod** [comment]
I think this should stay open as long as the steps to reproduce listed by @boombata above (https://github.com/salesagility/SuiteCRM/issues/1491#issuecomment-273205561) still reproduce the bug in the latest versions (I think they do)...

**Comment by eggsurplus** [comment]
I think I have a solution for this. Will open a PR, but requesting help on testing to make sure it meets your expectations.

**Comment by pgorod** [comment]
Thanks Jason! (I managed not to call you [Json](https://github.com/salesagility/SuiteCRM/issues/5028#issuecomment-502015386) like some other devs around here!)

I'm sure a solution for this will have many fans in this thread and we will surely help test it. Thanks.

**Comment by eggsurplus** [comment]
:) I appreciate that and the help.

**Comment by eggsurplus** [comment]
Added a PR that includes a fix for this: https://github.com/salesagility/SuiteCRM/pull/7405

The Security Groups block in SugarFeed->get_list_view_data() is no longer required and has been removed. It has been replaced in SugarFeedDashlet->process() with logic that removes the row if the current user does not have ListView access to that record.

**Comment by Mac-Rae** [comment]
This issue has been closed as version 7.8.x of SuiteCRM has reached its end of life and therefore will no longer be receiving updates. We have also been unable to replicate this issue in any newer version of SuiteCRM such as 7.10.x or 7.11.x however if you are able, please leave a comment explaining how and we'll re-open this issue.

For more information about 7.8.x end of life please visit our [blog post](https://suitecrm.com/7-8-x-end-of-life/)

**Comment by pgorod** [comment]
I didn't test it, but I believe #7474 was supposed to solve this issue, so it can be considered closed, not just "deprecated".
